### PR TITLE
Change the semantics of print

### PR DESCRIPTION
**Pull request by dhil**
This PR changes the semantics of the built-in function `print` such that it no longer emits a newline character. A new function `println : (String) ~> ()` has been added to the prelude which prints a string followed by a newline character.

In addition, the lexer now recognises `\b` as an escape character.

Finally, I have patched all the examples in the `examples/` directory to conform with the new semantics of `print`.

### CODE REVIEW

File: examples/handlers/aop.links [hunk 1]
```diff
@@ -6,7 +6,7 @@ sig runState :
         (Comp ({Get:(()) {}-> s,Put:(s) {}-> ()|e}, a)) ->
   (s) -> Comp ({Get{_},         Put{_}         |e}, (a, s))
 fun runState(m)(s)() {
-  (open handle(m) {
```
Comment (frank-emrich) [inline examples/handlers/aop.links]: it looks like you committed some unrelated other changes here

File: examples/handlers/aop2.links [hunk 2]
```diff
@@ -1,12 +1,12 @@
 typename Zero = [||];
 
-typename Comp(e::Row,a) = (() ~e~> a);
```
Comment (frank-emrich) [inline examples/handlers/aop2.links]: same in this file

File: examples/handlers/number_games.links [hunk 3]
```diff
@@ -0,0 +1,65 @@
+# Classic guess a number game
+typename Comp(e::Eff,a) = () ~e~> a;
+typename Answer = [|Low|Correct|High|];
+
```
Comment (frank-emrich) [inline examples/handlers/number_games.links]: added new example here?

File: examples/sessions/Elvina/IRC_Test.links [hunk 4]
```diff
@@ -117,20 +117,20 @@ fun connectToIRCServer() {
...
     # End of tests.
     writeToSocket("QUIT\n", rs);
-    print("QUIT");
+    println("QUIT");
     closeSocket(rs);
     ()
 }
 
-# Writes LinksCalc: and given expression to socket; prints it.
+# Writes LinksCalc: and given expression to socket; printlns it.
```
Comment (frank-emrich) [inline examples/sessions/Elvina/IRC_Test.links]: your regex should have been something like print[^a-zA-Z] to prevent breaking comments containing "print"

File: examples/handlers/aop.links (hunk 1 above)
Comment (dhil) [reply examples/handlers/aop.links]: Yes, I fixed a broken example too.

File: examples/handlers/number_games.links (hunk 3 above)
Comment (dhil) [reply examples/handlers/number_games.links]: Apparently, yes... This is the example from my ProWeb talk.

File: examples/sessions/Elvina/IRC_Test.links (hunk 4 above)
Comment (dhil) [reply examples/sessions/Elvina/IRC_Test.links]: Fixed.
